### DISCUSSION TITLE
feature request: support for websocket streaming protocol

### DISCUSSION THREAD

**Discussion by or-shachar** [community/community#54846] [Enterprise]
### Select Topic Area

Question

### Body

**TL;DR: I'm looking for a way to get github events using websocket streams.**

---

Today if an organization wants to get updates from Github it has two options IIUC:
1. **Polling API calls** - Using various list APIs - get new events (pro: no need to expose public endpoint. con: needs special care for each event type)
2. **Listening to webhook** - Create a webhook endpoint and register it (pro: easy subscription to any event. con: requires public endpoint).

Many enterprise organizations would want to subscribe to any event while avoiding the public endpoint.

Slack solved that by offering a socket connection - https://api.slack.com/apis/connections/socket that serves same events like the webhook but is based on a persistent connection between the client and the server (with an idempotent reconnect protocol).

I've searched everywhere and couldn't find something of that sort in Github. I think it would be an amazing addition to the developers' ecosystem.
